app/src/main/java/com/github/se/orator/ui/settings/SettingsScreen.kt

```diff
@@ -165,15 +120,27 @@ fun SettingsScreen(
                     titleContentColor = MaterialTheme.colorScheme.onSurface,
                     actionIconContentColor = MaterialTheme.colorScheme.onSurfaceVariant,
                 ))
-      },
-      content = { padding ->
-        Column(
+      }) { padding ->
+        LazyColumn(
             modifier = Modifier.fillMaxSize().padding(padding).testTag("settingsScreen"),
             verticalArrangement = Arrangement.spacedBy(AppDimensions.spacerWidthMedium)) {
-              for (setting in listOfSettings) {
-                TextButtonFun(setting, themeViewModel)
-                HorizontalDivider(thickness = AppDimensions.dividerThickness)
+
+              // Permissions
+              item {
+                TextButtonFun(listOfSettings[0]) {
+                  context.startActivity(
+                      Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS).apply {
+                        data = Uri.fromParts("package", context.packageName, null)
+                      })
+                }
               }
+              item {
+                TextButtonFun(listOfSettings[1]) {
+                  themeViewModel?.switchTheme()
+                  Log.d("SettingsScreen", "Theme switch")
+                }
+              }
+              item { TextButtonFun(listOfSettings[2]) { Log.d("SettingsScreen", "About") } }
```

Comment:
This code works well and achieves its purpose, but using an enum would improve modularity by allowing you to iterate through entries in a loop. Each enum entry could include fields such as a `display name`, an `icon`, and an `onClick` action, making the code more scalable—especially in cases where you need to handle multiple items (e.g., 10 icons) without duplicating similar code.

Reply:
Hey, this was the case before. The problem is that the `onClick` function sometimes need values that are passed as parameters in the `SettingsScreen.kt` composable. As theses values differ for each `onClick` function, the whole point of creating an enum would be defeated, as the code would be even more complicated.
If you are thinking of a way to create an enum by bypassing this issue of needing these specific variables accessible only from the composable function, please share it, I would be glad to add it.

Reply:
Yes you're right. I hadn't realized how complicated it would actually be to implement the custom on click functions for each context. My bad for flagging this.